can/interfaces/socketcan/socketcan.py

```diff
@@ -734,12 +705,14 @@ def _send_periodic_internal(
 
         msgs_channel = str(msgs[0].channel) if msgs[0].channel else None
         bcm_socket = self._get_bcm_socket(msgs_channel or self.channel)
-        # TODO: The SocketCAN BCM interface treats all cyclic tasks sharing an
-        # Arbitration ID as the same Cyclic group. We should probably warn the
-        # user instead of overwriting the old group?
-        task = CyclicSendTask(bcm_socket, msgs, period, duration)
+        task_id = self._get_next_task_id()
+        task = CyclicSendTask(bcm_socket, task_id, msgs, period, duration)
         return task
 
+    def _get_next_task_id(self) -> int:
+        self._task_id = (self._task_id + 1) % (2 ** 11 - 1)
```

Comment:
I believe you still need mutual exclusion around the `task_id` since access to the `Bus` via `send_periodic` isn't serialized. I think we assume that `send_periodic` can be called concurrently.

Reply:
I am not really sure in that. There are ThreadSafeBus, but I don't really understand whether it serializes access to periodic messages sending. Anyway, I've added a guard.